### PR TITLE
Directory support for rv packages

### PR DESCRIPTION
**Pull request by shannonyearwood**
### Summarize your change.
When creating a package you can list folders with files that should be installed in the SupportPath Plugins folder rather than listing each file individually

### Describe the reason for the change.
It should be easier/quicker to create packages

### Describe what you have tested and on which operating system.
MacOS, Windows, Linux

### Add a list of changes, and note any that might need special attention during the review.
Add a list of changes, and note any that might need special attention during the review.
Packages are zipped with directory structure
All files in package and its directories are zipped
The app will traverse the PACKAGE file and see the list of folders specified. Based on this, all files under the folder and its subdirectories will be added to location provided.
N.B. When creating PACKAGE file if you want to use new behaviour it takes the following format:
folders:

folder: [FolderName]
location: [LocationName]

### CODE REVIEW

**Comment by geffrak** [comment]
Also, can you pull the ASWF OpenRV main branch in your branch (using --rebase) to eliminate the already merged commits?

File: src/lib/app/RvPackage/PackageManager.cpp [hunk 1]
```diff
@@ -1070,6 +1106,26 @@ PackageManager::loadPackageInfo(const QString& infileNonCanonical)
 				yaml_event_delete(&input_event);
 				yaml_parser_delete(&parser);
 
+        for (auto& auxFolder : package.auxFolders) {
```
Comment (shannonyearwood) [inline src/lib/app/RvPackage/PackageManager.cpp:1109]: @geffrak This code block is the difference. Removed the use of the QT native separators

Comment (geffrak) [reply src/lib/app/RvPackage/PackageManager.cpp:1109]: Ok great!

File: src/lib/app/RvPackage/PackageManager.cpp [hunk 2]
```diff
@@ -948,9 +964,11 @@ PackageManager::loadPackageInfo(const QString& infileNonCanonical)
 				package.openrvversion = "1.0.0";
 				bool modeState = false;
 				bool auxFileState = false;
+        bool auxFolderState = false;
```
Comment (bernie-laberge) [inline src/lib/app/RvPackage/PackageManager.cpp:967]: Missing indentation ?

Comment (shannonyearwood) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: Very strange, the indentation is there on my local machine

Comment (bernie-laberge) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: That's because this file is filled with tabs (not your fault : legacy !)

Comment (shannonyearwood) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: Should I reformat the file to only use one (tabs or spaces)?

Comment (bernie-laberge) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: Are you sure you set your visual studio code to the right number of spaces per tab ?
It should be:
    "editor.insertSpaces": true,
    "editor.tabSize": 4,
https://code.visualstudio.com/docs/editor/codebasics#_indentation

Comment (shannonyearwood) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: I'm using CLion but the tab size is already set to 4

Comment (bernie-laberge) [reply src/lib/app/RvPackage/PackageManager.cpp:967]: If you could that would be great : We prefer spaces. NOT tabs.

Comment (shannonyearwood) [reply src/lib/app/RvPackage/PackageManager.cpp]: Fixed

Comment (bernie-laberge) [reply src/lib/app/RvPackage/PackageManager.cpp]: Thank you !!!

**Review by geffrak — APPROVED** [review]

**Review by bernie-laberge — APPROVED** [review]

**Pull request merged**